2.deploy_RShiny/NIH_grants_app/server.r

```diff
@@ -0,0 +1,118 @@
+library(shiny)
+library(ggplot2)
+library(dplyr)
+library(arrow)
+library(shinyWidgets)
+
+# Install required packages if not already installed
+if (!require(ggExtra)) {
+  install.packages("ggExtra")
+}
+if (!require(patchwork)) {
+  install.packages("patchwork")
+}
+if (!require(ggplotify)) {
+  install.packages("ggplotify")
+}
+if (!require(rsconnect)) {
+  install.packages("rsconnect")
+}
+
+
+library(patchwork)
+library(ggExtra)
+library(ggplotify)
+library(rsconnect)
+
+# Define server logic
+server <- function(input, output, session) {
+  df <- arrow::read_parquet("All_academic_projects_funded_by_NIH_cleaned.parquet")
```

Comment:
Seeing how this file gets used and depending on how data reads impact the load time for the dashboard, consider using LZ4 compression for the parquet file instead of Snappy (I'm guessing that Snappy was used by default). Generally this might result in a larger file but faster read times.